### PR TITLE
[REF] web_editor: prevent duplicate code in LinkDialog and LinkTools

### PR DESCRIPTION
**Pull request by Zynton**
LinkTools were recently introduced to provide a new UI from link editing within a sidebar where there is one. It was implemented separately from the existing LinkDialog despite them having a lot of code in common. In order to prevent future discrepancies between them, this commit introduces the Link widget which contains the common logic for link editing, to be extended by LinkTools and LinkDialog (via an intermediary widget).

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web_editor/static/src/js/wysiwyg/widgets/link.js [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+        });
+        this.$link.attr(attrs);
+        if (!this.$link.attr('target')) {
+            this.$link[0].removeAttribute('target');
+        }
+        if (data.content !== this.data.originalText || data.url !== this.data.url) {
+            const content = (data.content && data.content.length) ? data.content : data.url;
+            // If there is a this.data.originalText, it means that we selected
+            // the text and we could not change the content through the text input.
+            // .html() in needed in case we selected images.
```
Comment (dmo-odoo) [inline addons/web_editor/static/src/js/wysiwyg/widgets/link.js:159]: tant qu'à faire autant corriger la typo ici

**Comment by dmo-odoo** [comment]
@odoo/rd-security This PR solves the code duplication that was introduced with one of the new features of the new editor. Unfortunately, one of the line that were moved triggered the security checks.

**Comment by mart-e** [comment]
@robodoo override=ci/security
Reason: `content` contains the part between `<a>` tags which can be html. The content is sanitized sever side and one can, at best, make a self-xss but will not impact others after saving.

**Comment by dmo-odoo** [comment]
@robodoo r+

**Pull request closed without merging**